ha/setup/ha_setup.py

```diff
@@ -353,9 +404,33 @@ def process(self):
         """
         Log.info("Processing cleanup command")
         try:
-            # Destroy the cluster
-            output = self._execute.run_cmd(const.PCS_CLUSTER_DESTROY, check_error=True)
-            Log.info(f"Cluster destroyed. Output: {output}")
+            node_name = self.get_node_name()
+            # Get a list of current cluster nodes
+            cluster_nodes = self._confstore.get(const.CLUSTER_CONFSTORE_NODES_KEY)
+            # Try to connect to these nodes and remove the current node.
+            node_removed = False
+            for key in cluster_nodes:
+                remote_node = key.split('/')[-1]
+                if remote_node != node_name:
+                    Log.info(f"Removing {node_name} using remote node: {remote_node}")
+                    remote_executor = SSHRemoteExecutor(remote_node)
+                    try:
+                        # TODO: Change following PCS command to CLI when available.
+                        remote_executor.execute(const.PCS_CLUSTER_NODE_REMOVE.replace("<node>", node_name))
+                        Log.info(f"Removed {node_name} from the cluster")
+                        node_removed = True
+                        break
+                    except Exception as e:
+                        Log.error(f"Removing {node_name} using remote node {remote_node} failed with error: {e}, \
+                                if available, will try with other node")
+
+            if node_removed == False:
+                Log.info(f"{node_name} remove failed/last node in the cluster, destroy the cluster")
+                output = self._execute.run_cmd(const.PCS_CLUSTER_DESTROY, check_error=True)
```

Comment:
Using node_removed flag for identifying the last node. I think, it will cause the problem if "remove node" fails for an intermediate node. We can identify the last node by count number of nodes present in conf_store.

Reply:
I think no. const.PCS_CLUSTER_DESTROY will destroy the cluster only on that specified node. The cluster on other nodes remains intact.
I think "remove node" is the best thing to try since it will remove the specified node from cluster. The "remove node" is to be tried from the other nodes in the cluster (similar to "add node"). So if somehow the "remove node" from all other node fails, then we try const.PCS_CLUSTER_DESTROY as a last resort. Also, if it is the last node being removed then we cannot do "remove node", hence do const.PCS_CLUSTER_DESTROY.

Reply:
> 
> 
> A general comment - please verify that all the operations are idempotent. If the command is executed for a node again, should not fail and do nothing if it has already done what it is supposed to do.

@ajaysrivas I think we are good with post_install and init already. Added some checks in config and cleanup.